c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -319,73 +319,97 @@ class TypeEncoder final : public TypeVisitor<TypeEncoder> {
+            // All the SVE types present in Clang 10 are 128-bit vectors
+            // (see `AArch64SVEACLETypes.def`), so we can divide 128 
+            // by their element size to get element count.
+            auto ElemCount = 128 / Context->getTypeSize(ElemType);
+#endif // CLANG_VERSION_MAJOR >= 11
+            auto ElemTypeTag = encodeQualType(ElemType);
+            encodeType(T, TagVectorType,
+                       [&](CborEncoder *local) {
+                           cbor_encode_uint(local, ElemTypeTag);
+                           cbor_encode_uint(local, ElemCount);
+                       });
+
+            VisitQualType(ElemType);
             return;
         }
-        #endif
-
-        // clang-format off
-        switch (kind) {
-        default: {
-            auto pol = clang::PrintingPolicy(Context->getLangOpts());
-            auto warning = std::string("Encountered unsupported BuiltinType kind ") +
-                           std::to_string((int)kind) + " for type " +
-                           T->getName(pol).str();
-            printWarning(warning, clang::FullSourceLoc());
-            tag = TagTypeUnknown;
-            break;
-        }
+#endif // CLANG_VERSION_MAJOR >= 10
 
-        case BuiltinType::BuiltinFn:  tag = TagBuiltinFn;   break;
-        case BuiltinType::UInt128:    tag = TagUInt128;     break;
-        case BuiltinType::Int128:     tag = TagInt128;      break;
-        case BuiltinType::Short:      tag = TagShort;       break;
-        case BuiltinType::Int:        tag = TagInt;         break;
-        case BuiltinType::Long:       tag = TagLong;        break;
-        case BuiltinType::LongLong:   tag = TagLongLong;    break;
-        case BuiltinType::UShort:     tag = TagUShort;      break;
-        case BuiltinType::UInt:       tag = TagUInt;        break;
-        case BuiltinType::ULong:      tag = TagULong;       break;
-        case BuiltinType::ULongLong:  tag = TagULongLong;   break;
-        case BuiltinType::Half:       tag = TagHalf;        break;
-        #if CLANG_VERSION_MAJOR >= 11
-        case BuiltinType::BFloat16:   tag = TagBFloat16;    break;
-        #endif
-        case BuiltinType::Float:      tag = TagFloat;       break;
-        case BuiltinType::Double:     tag = TagDouble;      break;
-        case BuiltinType::LongDouble: tag = TagLongDouble;  break;
-        case BuiltinType::SChar:      tag = TagSChar;       break;
-        case BuiltinType::UChar:      tag = TagUChar;       break;
-        case BuiltinType::Char_U:     tag = TagChar;        break;
-        case BuiltinType::Char_S:     tag = TagChar;        break;
-        case BuiltinType::Void:       tag = TagVoid;        break;
-        case BuiltinType::Bool:       tag = TagBool;        break;
-        case BuiltinType::WChar_S:    tag = TagSWChar;      break;
-        case BuiltinType::WChar_U:    tag = TagUWChar;      break;
-        }
-        // clang-format on
+        const TypeTag tag = [&] {
```

Comment:
Is there anything changed semantically here? Looks more like a style change, these should generally be separate patches but I'll allow this one.

Reply:
I don't think dbe85a1f36dc4f1145e2fd345fd39a34375b88c4 changes any semantics, no.

Reply:
No, that part was just a pure refactor to make the code a bit more concise.